mountpoint-s3/src/cli.rs

```diff
@@ -895,6 +896,7 @@ where
         (None, Some((config, bucket_name, cache_bucket_name))) => {
             tracing::trace!("using S3 Express One Zone bucket as a cache for object content");
             let express_cache = ExpressDataCache::new(client.clone(), config, bucket_name, cache_bucket_name);
+            block_on(express_cache.verify_cache_valid())?;
```

Comment:
how does the error looks in the log? would it be useful to add some context about the put operation? (similar to [ListObjectsV2](https://github.com/awslabs/mountpoint-s3/blob/main/mountpoint-s3/src/cli.rs#L1061))
```rust
.with_context(|| format!("initial PutObject failed for bucket {cache_bucket_name}"))
```

Reply:
With the additional context, it looks like this:
```
2024-11-18T14:17:14.010171Z  WARN put_object_single{id=1 bucket="..." key="5f74d6e14553da6f2273143c53378cc2b8942ef97ff06bbc01756abc0c4bd6c9/_mountpoint_cache_metadata"}: mountpoint_s3_client::s3_crt_client: duration=4.885103ms request_id=RN7K3ZYAH0Z72YS5 request_result=MetaRequestResult { response_status: 400, crt_error: Error(14343, "aws-c-s3: AWS_ERROR_S3_INVALID_RESPONSE_STATUS, Invalid response status from request"), error_response_headers: Some(Headers { inner: 0x7f80c00172c0 }), error_response_body: Some("<?xml version=\"1.0\" encoding=\"UTF-8\"?>\n<Error><Code>InvalidStorageClass</Code><Message>The storage class you specified is not valid</Message><StorageClassRequested>EXPRESS_ONEZONE</StorageClassRequested><RequestId>RN7K3ZYAH0Z72YS5</RequestId><HostId>3TlUgk4EOMSHtVRavhSrsoorc/bi6mpLOU0yuw96IK7cRzVFOxL5KbsfyTQUHIzReLqDH5tonuk=</HostId></Error>") } meta request failed
Error: initial PutObject failed for shared cache bucket ...

Caused by:
    0: Client error
    1: Unknown response error: MetaRequestResult { response_status: 400, crt_error: Error(14343, "aws-c-s3: AWS_ERROR_S3_INVALID_RESPONSE_STATUS, Invalid response status from request"), error_response_headers: Some(Headers { inner: 0x7f80c00172c0 }), error_response_body: Some("<?xml version=\"1.0\" encoding=\"UTF-8\"?>\n<Error><Code>InvalidStorageClass</Code><Message>The storage class you specified is not valid</Message><StorageClassRequested>EXPRESS_ONEZONE</StorageClassRequested><RequestId>RN7K3ZYAH0Z72YS5</RequestId><HostId>3TlUgk4EOMSHtVRavhSrsoorc/bi6mpLOU0yuw96IK7cRzVFOxL5KbsfyTQUHIzReLqDH5tonuk=</HostId></Error>") }
```